frontend/src/pages/CompareV2.tsx

```diff
@@ -15,9 +15,95 @@
  */
 
 import React from 'react';
+import { useQuery } from 'react-query';
+import { ApiRunDetail } from 'src/apis/run';
+import { QUERY_PARAMS } from 'src/components/Router';
 import { commonCss } from 'src/Css';
+import { Apis } from 'src/lib/Apis';
+import { URLParser } from 'src/lib/URLParser';
+import { errorToMessage } from 'src/lib/Utils';
+import {
+  getExecutionsFromContext,
+  getKfpV2RunContext,
+  getOutputLinkedArtifactsInExecution,
+  LinkedArtifact,
+} from 'src/mlmd/MlmdUtils';
+import { Execution } from 'src/third_party/mlmd';
+import { PageProps } from './Page';
+
+interface ExecutionArtifacts {
+  execution: Execution;
+  linkedArtifacts: LinkedArtifact[];
+}
+
+interface RunArtifacts {
+  run: ApiRunDetail;
+  executionArtifacts: ExecutionArtifacts[];
+}
+
+function CompareV2(props: PageProps) {
+  const queryParamRunIds = new URLParser(props).get(QUERY_PARAMS.runlist);
+  const runIds = (queryParamRunIds && queryParamRunIds.split(',')) || [];
+
+  // Retrieves run details.
+  const { data: runs } = useQuery<ApiRunDetail[], Error>(
+    ['run_details', { ids: runIds }],
+    () => Promise.all(runIds.map(async id => await Apis.runServiceApi.getRun(id))),
+    {
+      staleTime: Infinity,
+      onError: async error => {
+        const errorMessage = await errorToMessage(error);
+        props.updateBanner({
+          additionalInfo: errorMessage ? errorMessage : undefined,
+          message: `Error: failed loading ${runIds.length} runs. Click Details for more information.`,
+          mode: 'error',
+        });
+      },
+      onSuccess: () => props.updateBanner({}),
+    },
+  );
+
+  // Retrieves MLMD states (executions and linked artifacts) from the MLMD store.
+  const { data: runArtifacts } = useQuery<RunArtifacts[], Error>(
+    ['run_artifacts', { runIds, runs }],
+    () => {
+      if (runs) {
+        return Promise.all(
+          runIds.map(async (r, index) => {
+            const context = await getKfpV2RunContext(r);
+            const executions = await getExecutionsFromContext(context);
+
+            const executionArtifacts = await Promise.all(
+              executions.map(async execution => {
+                const linkedArtifacts = await getOutputLinkedArtifactsInExecution(execution);
+                return {
+                  execution,
+                  linkedArtifacts,
+                } as ExecutionArtifacts;
+              }),
+            );
+
+            return {
+              run: runs[index],
+              executionArtifacts,
+            };
+          }),
+        );
+      }
+      return [];
+    },
+    {
+      staleTime: Infinity,
+      onError: error =>
+        props.updateBanner({
+          message: 'Cannot get MLMD objects from Metadata store.',
+          additionalInfo: error.message,
```

Comment:
I used `error.message` here rather than `await errorToMessage(error)` as this code segment was referenced from the following, and seems to work well (see the tests): https://github.com/kubeflow/pipelines/blob/9d5433279e4cf96ad4429a7509ab0890065c32ff/frontend/src/pages/RunDetailsV2.tsx#L112-L122.